integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -576,19 +584,27 @@ fn get_memtrie_for_shard(
     memtrie
 }
 
+// TODO: fix FlatStorageChunkView::iter_range() and remove the warn_only argument, just panicking in all cases
```

Comment:
What if instead of adding warnings we skip the check if flat state head is not at the same level with memtries?

Example: 
```
        if let FlatStorageStatus::Ready(status) = client
            .chain
            .runtime_adapter
            .get_flat_storage_manager()
            .get_flat_storage_status(shard_uid)
        {
            if status.flat_head.hash != final_head.prev_block_hash {
                tracing::warn!(target: "test", "skipping flat storage - memetrie state check");
                continue;
            } else {
                tracing::debug!(target: "test", "checking flat storage - memetrie state");
            }
        } else {
            continue;
        };
```
before calling `chunk_view()` in `assert_state_sanity`

Reply:
Proposed PR: https://github.com/near/nearcore/pull/12592

(with memtrie spelled correctly! 🤣 )

Reply:
reverted in favor of your PR

Reply:
somebody should still fix `FlatStorageChunkView::iter_range()` one day tho